Searching Algorithms/Jump Search Algorithm/C++/JumpSearch.cpp

```diff
@@ -0,0 +1,52 @@
+/* ===== ===== =====
+
+Theory of Programming
+
+Jump Search Algorithm
+http://theoryofprogramming.com/2016/11/10/jump-search-algorithm/
+GitHub - https://github.com/VamsiSangam/theoryofprogramming
+Code Contributor - Praveen Mylavarapu (saipraveen.ml)
+
+===== ===== ===== */
+
+#include <iostream>
+#include <math.h>
+
+using namespace std;
+
+int jumpSearch(int arr[], int n, int value)
+{
+    int i, right, left, jump = (int) sqrt(n);
+
+    // Keep jumping until value is between any two elements of the array or when i goes out of bound
+    for (i = 0; i + jump < n && arr[i] < value; i += jump);
+
+    if (value <= arr[i])    // right is i if search value is less than any jumped element in the array. If not, right is last element
+        right = i;
+    else
+        right = n;
+
+    left = right - jump;
```

Comment:
Use `left = max(right - jump, 0)` instead. Also add the comment `// ensuring left doesn't go negative` next to it.